pkg/controller/combo_controller.go

```diff
@@ -2,36 +2,183 @@ package controller
 
 import (
 	"context"
+	"fmt"
+	"strings"
+	"time"
 
 	"github.com/go-logr/logr"
+	"github.com/operator-framework/combo/api/v1alpha1"
+	combinationPkg "github.com/operator-framework/combo/pkg/combination"
+	templatePkg "github.com/operator-framework/combo/pkg/template"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/errors"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
-
-	"github.com/operator-framework/combo/api/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
 type combinationController struct {
 	client.Client
 	log logr.Logger
 }
 
-func (c *combinationController) manageWith(mgr ctrl.Manager, version int) error {
-	c.log = c.log.V(version)
+// manageWith creates a new instance of this controller
+func (c *combinationController) manageWith(mgr ctrl.Manager, verbosity int) error {
+	c.log = c.log.V(verbosity)
+	templateHandler := handler.EnqueueRequestsFromMapFunc(c.mapTemplateToCombinations)
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.Combination{}).
+		Watches(&source.Kind{Type: &v1alpha1.Template{}}, templateHandler).
 		Complete(c)
 }
 
+// mapTemplateToCombinations is responsible for taking the template object and finding all associated
+// combinations that should be requeued. This should only happen whenever a template is changed in someway.
+func (c *combinationController) mapTemplateToCombinations(template client.Object) []reconcile.Request {
+	if template == nil {
+		return nil
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	requests := []reconcile.Request{}
+
+	// Build the label selector for querying
+	templateName := template.GetName()
+	templateSelector, err := labels.Parse("ReferencedTemplate=" + templateName)
+	if err != nil {
+		return requests
+	}
+
+	// Find all of the combinations that rely on this template and enqueue them for updates
+	combinationList := v1alpha1.CombinationList{}
+	if err := c.List(ctx, &combinationList, &client.ListOptions{LabelSelector: templateSelector}); err != nil {
+		return requests
+	}
+
+	for _, combination := range combinationList.Items {
+		c.log.Info(fmt.Sprintf("enqueueing %s combination in response to associated %s template being updated", combination.Name, templateName))
```

Comment:
This probably doesn't need to be at log level, it could create a lot of logs in production environments. The requeued combination will be logged by the controller anyways.

Reply:
As it currently stands this won't be logged unless the verbosity is increased. The default verbosity of the controller is defined to not log. However, there is also a ticket that exists to rethink the logging strategy to be more effective which you can find in #62.